src/libopensc/card-gpk.c

```diff
@@ -242,7 +243,7 @@ match_path(sc_card_t *card, unsigned short int **pathptr, size_t *pathlen,
 
 	/* Make sure path starts with MF.
 	 * Note the cached path should always begin with MF. */
-	if (ptr[0] != GPK_FID_MF || curptr[0] != GPK_FID_MF)
+	if (ptr[0] != (GPK_FID_MF >> 8) || curptr[0] != (GPK_FID_MF >> 8))
```

Comment:
I think this change requires some more description what is changed and why and why it fixes your issue.